instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcSpanStatusExtractor.java

```diff
@@ -28,11 +42,35 @@ public void extract(
       }
     }
     if (status != null) {
-      if (!status.isOk()) {
+      if (isError.test(status)) {
         spanStatusBuilder.setStatus(StatusCode.ERROR);
       }
     } else {
       SpanStatusExtractor.getDefault().extract(spanStatusBuilder, request, status, error);
     }
   }
+
+  private static final Set<Status.Code> serverErrorStatuses = new HashSet<>();
+
+  static {
+    serverErrorStatuses.addAll(
+        Arrays.asList(
+            Status.Code.UNKNOWN,
+            Status.Code.DEADLINE_EXCEEDED,
+            Status.Code.UNIMPLEMENTED,
+            Status.Code.INTERNAL,
+            Status.Code.UNAVAILABLE,
+            Status.Code.DATA_LOSS));
+  }
+
+  private static boolean isServerError(Status status) {
+    return serverErrorStatuses.contains(status.getCode());
+  }
+
+  private static boolean isClientError(Status status) {
+    return !status.isOk();
+  }
+
+  @Immutable
```

Comment:
We don't really use this annotation very often; and it doesn't really make sense to mark interface as immutable

Reply:
this is to please errorprone, alternative is to use `@SuppressWarnings` to silence the enum is not immutable check